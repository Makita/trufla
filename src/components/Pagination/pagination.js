import React, { PureComponent } from 'react';

import { Pager } from 'react-bootstrap';

interface PaginationProps {
  page: number,
  handlePageChange: (page) => void
}

export default class Pagination extends PureComponent<PaginationProps, {}> {
  render() {
    const { page, handlePageChange } = this.props;
    const prevPage = page - 1;
    const nextPage = page + 1;

    return (
      <Pager>
        <Pager.Item previous disabled={prevPage <= 0} onClick={() => handlePageChange(prevPage)}>
          &larr; Previous Page
        </Pager.Item>
        <Pager.Item next onClick={() => handlePageChange(nextPage)}>
          Next Page &rarr;
        </Pager.Item>
      </Pager>
    );
  }
}