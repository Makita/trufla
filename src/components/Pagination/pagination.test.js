/* eslint-disable no-undef */
import * as React from 'react';
import Pagination from './pagination';

describe("Pagination", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Pagination />);

    expect(wrapper).toMatchSnapshot();
  })
});