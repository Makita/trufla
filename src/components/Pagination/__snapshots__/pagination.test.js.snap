// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<Pager
  bsClass="pager"
>
  <PagerItem
    disabled={false}
    next={false}
    onClick={[Function]}
    previous={true}
  >
    ← Previous Page
  </PagerItem>
  <PagerItem
    disabled={false}
    next={true}
    onClick={[Function]}
    previous={false}
  >
    Next Page →
  </PagerItem>
</Pager>
`;
